Firmware/menu.cpp

```diff
@@ -65,7 +65,7 @@ void menu_goto(menu_func_t menu, const uint32_t encoder, const bool feedback, bo
 
 void menu_start(void)
 {
-    if (lcd_encoder > 0x8000) lcd_encoder = 0;
+    if (lcd_encoder > INT8_MAX) lcd_encoder = 0;
```

Comment:
This change is a bit concerning. The SD card menu can easily exceed 127 items.

Reply:
Is this really the total number of menu items?
Anyway, we can change it to `INT16_MAX`

Reply:
I may be wrong, but I think that line can be removed entirely. I don't really see the point of it. The intention seems to be to check that the lcd_encoder doesn't get too large, in which case it's reset to 0. That's all nice and good, except now that lcd_encoder is a int16_t, the range of lcd_encoder is reduced. So the maximum positive value it can have is 0x7fff. If that check is removed entirely, then lcd_encoder is allowed to overflow and become negative, something which is checked a bit lower in the code and it's handled almost the same way as the code I'm proposing to remove. This should work around the issue for menus with many items, while still working more or less the same as before.

Reply:
ok, makes sense ... I'll remove the line